controllers/utils/mapper.go

```diff
@@ -92,3 +99,97 @@ func NHCByRemediationCRMapperFunc(logger logr.Logger) handler.MapFunc {
 	}
 	return delegate
 }
+
+// MHCByNodeMapperFunc return the Node-to-MHC mapper function
+func MHCByNodeMapperFunc(c client.Client, logger logr.Logger) handler.MapFunc {
+	delegate := func(o client.Object) []reconcile.Request {
+		requests := make([]reconcile.Request, 0)
+
+		node := &v1.Node{}
+		if err := c.Get(context.Background(), client.ObjectKey{Name: o.GetName()}, node); err != nil {
+			if !errors.IsNotFound(err) {
+				node.Name = o.GetName()
```

Comment:
Maybe I'm missing something, but I think there is a panic risk here :thinking: .
(i.e after node retrieval was unsuccessful we are trying to access a Node field)

Reply:
> I think there is a panic risk here

looking at the implementation, the node pointer isn't overwritten with nil when an error occurs. And it's the same in the original code (https://github.com/openshift/machine-api-operator/blob/6cdf6f48afac36d0e6e800e42838531f0fca433e/pkg/controller/machinehealthcheck/machinehealthcheck_controller.go#L568). However, that's not documented behavior, so yes, better be on the safe side, good catch 👍🏼 

> Also expecting a return after the error

no, we need continue to get the Machine and MHC just by node name, the node doesn't have to exist, it can be just deleted. We can still healthycheck the Machine in this case. Just like in the original MHC controller.

Reply:
>the node doesn't have to exist

true, but in case the node does not exist I'd expect `errors.IsNotFound(err)` to be true.
In case (`err != nil && !errors.IsNotFound(err)`) an error occurred and it is not a `not found` error IIUC this is indication of a problem.
The original code return in this case with a `nil` but TBH i didn't look to deeply in the original code so I'm sure what it means.

Reply:
Ah, you are in the else block. I do return there with the empty `requests` slice. Has the same effect as `nil` (not adding anything to the queue), I just aligned it to existing mapper functions in order to be consistent. The function signature we have to use here doesn't allow to return an error.

Reply:
>Ah, you are in the else block

no I was refering to the `if` block and the `!` in particular.

IIUC The original code is equivalent to:
```
if errors.IsNotFound(err) {
  node.Name = o.GetName()
 ... 
```	
while current code is:
```
if !errors.IsNotFound(err) {
  node.Name = o.GetName()
 ... 
```

Reply:
oooohhhh... good catch! 🙈